Allow users to pick local audio source location

ellieglad opened the pull request:
`context.getExternalFilesDir(null)` will default to what it believes is the primary storage, sometimes that's internal and sometimes that's external. With this change users will now be able to pick which option they want. If the user does nothing it defaults back to the existing `getExternalFilesDir` and allows the system to pick the default.

The new option shown here under "Choose local audio directory location".  
![image](https://github.com/user-attachments/assets/c493cf0d-b39e-4338-82c2-fbea6bf1c7eb)  

Open to any changes or feel free to close if it's believed that this will not add value.
Thanks!

ellieglad committed:
feat: allow users to pick local audio source location

context.getExternalFilesDir(null) will default to what it believes is the primary storage, sometimes that's internal and sometimes that's external. With this change users will now be able to pick which option they want. If the user does nothing it defaults back to the existing getExternalFilesDir and allows the system to pick the default.

KamWithK commented on app/src/main/java/com/kamwithk/ankiconnectandroid/routing/LocalAudioAPIRouting.java:
```diff
@@ -76,7 +78,12 @@ public LocalAudioAPIRouting(Context context) {
 
     private EntriesDatabase getDB() {
         // TODO global instance?
-        File databasePath = new File(context.getExternalFilesDir(null), "android.db");
+        //File databasePath = new File(context.getExternalFilesDir(null), "android.db");
```
Why do we keep the dead code here, remove unless it's for good reason and if it is it needs a comment above

ellieglad replied:
Removed it, it was a leftover from testing.

KamWithK commented on app/src/main/java/com/kamwithk/ankiconnectandroid/SettingsActivity.java:
```diff
@@ -63,7 +66,9 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
                         Toast.makeText(getContext(), "Cannot get local audio folder, as context is null.", Toast.LENGTH_LONG).show();
                     } else {
-                        Toast.makeText(getContext(), "Local audio folder: " + context.getExternalFilesDir(null), Toast.LENGTH_LONG).show();
+                        String folder = context.getSharedPreferences(getString(R.string.shared_app_preferences), 0)
+                                .getString("storage_location", context.getExternalFilesDir(null).getAbsolutePath());
```
Surely there is a nicer way we can do this?

ellieglad replied:
Refactored the code to simply use `getDefaultSharedPreferences` instead of having to create a new one.

ellieglad committed:
refactor: use default SharedPreferences

This change makes the code a lot cleaner, relying on the default shared preferences instead of trying to sync new ones.

ellieglad committed:
chore: fix some unfinished text

KamWithK commented:
@ellieglad would you be able to rebase the latest changes into your branch?

ellieglad committed:
Merge branch 'KamWithK:master' into master

ellieglad commented:
@KamWithK done. github's sync fork option only allows for merging not rebasing unfortunately.

KamWithK commented:
<img width="546" height="1235" alt="image" src="https://github.com/user-attachments/assets/a8eeecfb-55fa-48b3-8884-b4d264ecd7a4" />

Hey so someone tried this PR for me, but they weren't able to choose a folder

Are you on Discord by any chance, it'd be easier to problem solve/discuss in messages

Discussions on TheMoeWay or DMs both work

ellieglad committed:
feat: allow user to choose custom local storage
